mesonbuild/compilers/asm.py

```diff
@@ -282,3 +283,47 @@ def get_crt_compile_args(self, crt_val: str, buildtype: str) -> T.List[str]:
 
     def depfile_for_object(self, objfile: str) -> T.Optional[str]:
         return None
+
+
+class MetrowerksAsmCompiler(MetrowerksCompiler, Compiler):
+    language = 'nasm'
+
+    def __init__(self, ccache: T.List[str], exelist: T.List[str], version: str,
+                 for_machine: 'MachineChoice', info: 'MachineInfo',
+                 linker: T.Optional['DynamicLinker'] = None,
+                 full_version: T.Optional[str] = None, is_cross: bool = False):
+        Compiler.__init__(self, ccache, exelist, version, for_machine, info, linker, full_version, is_cross)
+
+        self.warn_args = {'0': [], '1': [], '2': [], '3': [], 'everything': []}  # type: T.Dict[str, T.List[str]]
+        self.can_compile_suffixes.add('s')
```

Comment:
The inherited MetrowerksCompiler would check for non-cross builds and error out, but we are not running `MetrowerksCompiler.__init__(self)` here.

(Which would set a couple options too specific for here, I guess. In fact, maybe we should factor that out into a MetrowerksCompiler and a MetrowerksClikeCompiler instead of setting warn_args back to empty?)

Reply:
A separate class just to deal with warn_args is overkill imo, considering also that the Metrowerks Assembler would be the only compiler using it as there aren't really any other Metrowerks-like compilers out there. For now I'm just going to call MetrowerksCompiler.__init__(self) to make CodeQL happy.